### PR TITLE
{packaging} Bump semver to 3.0.0

### PR DESCRIPTION
**Pull request by major**
**Related command**
N/A

**Description**
Many Linux distributions, including Fedora, have updated to semver 3.0.0. It would be helpful if azure-cli could use this newer version.

**Testing Guide**
No testing changes needed.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
packaging

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 1]
```diff
@@ -129,7 +129,7 @@ python-dateutil==2.8.0
 requests-oauthlib==1.2.0
 requests[socks]==2.26.0
 scp==0.13.2
-semver==2.13.0
+semver==3.0.0
```
Comment (jiasli) [inline src/azure-cli/requirements.py3.Darwin.txt:132]: The dependency on `semver` was introduced by #16857.

**Review by bebound — APPROVED** [review]

**Comment by jiasli** [comment]
Thanks @major for your contribution. However, we decided we will drop the `semver` dependency and use `packaging.version` instead.

**Comment by jiasli** [comment]
@major, as shown in https://github.com/Azure/azure-cli-extensions/pull/6575#discussion_r1278847081 and https://github.com/Azure/azure-cli/pull/26797#discussion_r1274420674 we would want to keep the dependency on `semver`.

It'll be great if you can recreate the branch and reopen this PR, so that we can give you credit for submitting this PR. Thanks!

**Comment by major** [comment]
@jiasli Will do. Thank you.

**Comment by major** [comment]
@jiasli I restored the branch and rebased the commit, but it doesn't look like I have permissions to reopen this PR. Could you do that?

If not, should I open a new PR and reference this one?

**Pull request closed without merging**